### PR TITLE
Add .gitkeep to ensure template directory is always present.

### PR DESCRIPTION
**Pull request by joeflack4**
## Changes
Bugfix: `run-command.sh`: `tmp/` nonexistent
If run-command.sh runs on a fresh clone, it will fail because it tries to append to a debug log, but it can't do that because the directory for that file doesn't exist.

## Additional details
Context: Encountered this when running a build on fresh clone of `mondo-ingest` (https://github.com/monarch-initiative/mondo-ingest/pull/443), which uses ODK.

### CODE REVIEW

**Review by matentzn — COMMENTED** [review]
Thank you very much @joeflack4 

Looks great at first glance, will take it for a spin later this week. Thanks a ton!

**Review by matentzn — COMMENTED** [review]
@gouttegd what do you think of this?

Basically we are trying to find a more general solution to the problem of non existing untracked but essential directories like `tmp`. (as an alternative to polluting the Makefile with hundres of `mkdir -p x` or adding the directories as dependencies of every goal.

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,12 @@
 target/
 bin/
 semantic.cache
-tmp/
+tmp/*
```
[the remark this answers is not in the document]
Comment (gouttegd) [reply .gitignore:4]: > I think that this fix requires updates to both

Why? The top-level `.gitignore` is only used for the ODK itself, it has nothing to do with the file that ends up in ODK-managed repositories.

**Review by gouttegd — COMMENTED** [review]
I have no objection to the general idea of using a file to ensure a directory such as `src/ontology/tmp/` always exists.

But I don’t like the choice of an empty `.gitkeep` file.

First, not all systems follow the convention of treating files whose name starts with a dot as hidden files. Windows at least certainly does not.

Second, instead of a (potentially hidden) empty file, I’d rather have a README file that actually explains what the folder is used for.

**Review by matentzn — CHANGES_REQUESTED** [review]
Ok, to go with @gouttegd suggestion:

Since `!src/ontology/tmp/README.md` already exists, in the template .gitignore lets

1. Add a `template/src/ontology/tmp/README.md` to the `tmp` dir instead of a `template/src/ontology/tmp/.gitkeep` (add some content like "This is a directory for  temporary files that are needed by the build process"
2. Reset the toplevel .gitignore the `master` branch state

Thank you!

**Comment by gouttegd** [comment]
The Makefile also needs updating to 1) remove all the explicit dependencies to `$(TMPDIR)` which would now be useless, and 2) more importantly, amend the `clean` rule, which would forcefully remove the `README` file along with all the other files in `$(TMPDIR)`:

```
rm -rf ./$(TMPDIR)/*
```

(1) can arguably be left to be done in a subsequent PR (those dependencies are useless but not harmful), but (2) is an immediate problem.

**Comment by gouttegd** [comment]
I would also like to see the approach being generalized to the other directories we need to keep (e.g. `mirror`). Again, this can arguably be done in a future PR but I think it would be better to do everything in one go here.

**Comment by gouttegd** [comment]
> more importantly, amend the clean rule, which would forcefully remove the README file along with all the other files in $(TMPDIR)

In fact, I think it would be much easier to stick with `.gitkeep`. Because the file is “hidden” on GNU/Linux (where the ODK runs), it would _not_ be removed by `rm -rf $(TMPDIR)/*`. Having a non-hidden file such as `README.md` would involve some convolutions to ensure that all files within `$(TMPDIR)` **except README.md** are removed.

Having a README.md file that explains what the directory is for would be nice but it is probably not worth the added complexity.

So, please disregard the comment about renaming `.gitkeep` to `README.md`.

**Review by matentzn — APPROVED** [review]
Thank you @joeflack4 and @gouttegd for finishing this up!

**Pull request merged**